lib/tar.ml

```diff
@@ -49,7 +46,12 @@ module Header = struct
     Int64.of_string tmp
 
   (** Unmarshal a string *)
-  let unmarshal_string (x: string) : string = trim_string x
+  let unmarshal_string (x: string) : string =
+    try
+      let first_0 = String.index x '\000' in
+      String.sub x 0 first_0
+    with
+      Not_found -> ""
```

Comment:
Interesting point, I tried to copy the earlier semantics, and think I failed slightly:
- originally, if `""` was passed, it was returned (since `Re.Str.split` returned `[]`)
- now, also if `"foo"` is passed, the empty string is returned.

I guess the right thing to do is to either return x or reformulate unmarshal (and others) with the possibility of failing.

Reply:
fixed in ea1f76a832909f22c93a9bd2000eb2d65fedb729